pwndbg/gdblib/typeinfo.py

```diff
@@ -130,10 +130,28 @@ def load(name: str) -> Optional[gdb.Type]:
         return None
 
 
+def cast_pointer(type: gdb.Type, addr: int):
+    """Create a gdb.Value at given address and cast it to the pointer of specified type"""
+    return gdb.Value(addr).cast(type.pointer())
+
+
+def get_typed_pointer(type_name: str, addr: int):
+    """Look up a type of type_name and return a gdb.Value of addr cast to that type"""
+    gdb_type = pwndbg.gdblib.typeinfo.load(type_name)
+    if gdb_type is None:
+        raise ValueError(f"Type '{type_name}' not found")
+    return cast_pointer(gdb_type, addr)
+
+
+def get_typed_pointer_value(type_name: str, addr: int):
+    """Read the pointer value of addr cast to type specified by type_name"""
+    return get_typed_pointer(type_name, addr).dereference()
+
+
+# FIXME: Consider deprecating this, as the name isn't as intuitive as get_typed_pointer()
 def read_gdbvalue(type_name: str, addr) -> gdb.Value:
     """Read the memory contents at addr and interpret them as a GDB value with the given type"""
-    gdb_type = pwndbg.gdblib.typeinfo.load(type_name)
-    return gdb.Value(addr).cast(gdb_type.pointer()).dereference()
```

Comment:
Huh, `read_gdbvalue` looks very similar to `poi`: https://github.com/pwndbg/pwndbg/blob/00adfbbb5e06564f7d351624e09dae898f485c57/pwndbg/gdblib/memory.py#L307

I agree that both names suck, and prefer your naming.

My only question is whether this is the right file to put these functions in, or is memory.py a better location?

Reply:
Hrm, `poi` has better type definitions than what I used so I'll fix mine up to use that. I don't like the name `poi` either. 

I'm happy to move these to `memory.py`. I think it makes sense given what else is in there, and that `poi` is more widely used, so people will likely expect them there. I guess the only real type-related part is the `load()` call used by `get_typed_pointer()`, and that will stay in `typeinfo.py`.

I can swap out all uses of `poi` and `read_gdbvalue` as part of this if you want, or can leave it for later.

Reply:
Phew I thought I was the only one who thought `poi` was a terrible name 😅

Reply:
I've replaced all uses of poi() and read_gdbvalue() with the new API, and moved it into gdblib/memory instead.

Reply:
I also dislike the `poi` name ;). I think it was taken from windbg or elsewhere...